### ISSUE TITLE
Produce fails to create app on AppStoreConnect with error Bundle ID already used

### ISSUE THREAD

**Issue by gibertoni** [fastlane/fastlane#16649]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
I am trying to create new Apps on AppStoreConnect using an script that I have and that worked many times in the past.

Right now, it is failing with the following error being thrown
```
[20:07:54]: Creating new app 'XXXXXX' on iTunes Connect
The `version` parameter is deprecated. Use `ensure_version!` method instead
/Users/guilhermeg/.rvm/gems/ruby-2.4.9/gems/fastlane-2.150.0.rc1/spaceship/lib/spaceship/tunes/tunes_client.rb:230:in `handle_itc_response': The Bundle ID you entered has already been used. The Bundle ID you entered has already been used. (Spaceship::Tunes::Error)
	from /Users/guilhermeg/.rvm/gems/ruby-2.4.9/gems/fastlane-2.150.0.rc1/spaceship/lib/spaceship/tunes/tunes_client.rb:325:in `create_application!'
	from /Users/guilhermeg/.rvm/gems/ruby-2.4.9/gems/fastlane-2.150.0.rc1/spaceship/lib/spaceship/tunes/application.rb:96:in `create!'
	from /Users/guilhermeg/.rvm/gems/ruby-2.4.9/gems/produce-2.0.0/lib/produce/itunes_connect.rb:25:in `create_new_app'
	from /Users/guilhermeg/.rvm/gems/ruby-2.4.9/gems/produce-2.0.0/lib/produce/itunes_connect.rb:12:in `run'
	from /Users/guilhermeg/.rvm/gems/ruby-2.4.9/gems/produce-2.0.0/lib/produce/manager.rb:8:in `start_producing'
	from create_itc_apps.rb:66:in `block in <main>'
	from create_itc_apps.rb:59:in `map'
	from create_itc_apps.rb:59:in `<main>'
```

It is as though the AppStoreConnect returned this error message. However I believe it may be due the latest update on AppStoreConnect and some part of our API that broke with it.

When I go to AppStoreConnect and create the App with same bundleID though the Web Interface, it works. So the Bundle ID is indeed valid. The error is misleading

I upgraded to 2.150.rc1 as mentioned on issue #16621 and #16639 to no avail.

##### Command executed
I execute a ruby script that runs the following
```
FastlaneCore::CommanderGenerator.new.generate(Produce::Options.available_options)
    Produce.config = FastlaneCore::Configuration.create(Produce::Options.available_options, options)
    puts Produce::Manager.start_producing
```

The options passed as argument are:
```
app_identifier: bundle_id,
    sku: bundle_id,
    app_name: "MyApp",
    app_version: "1.0.0",
    language: 'English',
    platform: "ios",
    team_id: 'XYZXYZ',
    itc_team_id: 00000,
    itc_team_name: 'My Company',
    skip_devcenter: true,
    username: username
```

### Environment

<details>
  <pre> ### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.14.5                                     |
| Ruby                        | 2.4.9                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.21.1 (Apple Git-122.3)        |
| Installation Source         | /usr/local/lib/ruby/gems/2.6.0/bin/fastlane |
| Host                        | Mac OS X 10.14.5 (18F203)                   |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.9/lib                |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 11.3.1                                      |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version     | Update-Status |
| -------- | ----------- | ------------- |
| fastlane | 2.150.0.rc1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| slack-notifier            | 2.3.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| gh_inspector              | 1.1.3        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| jwt                       | 2.1.0        |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.38.0       |
| emoji_regex               | 1.0.1        |
| jmespath                  | 1.4.0        |
| aws-partitions            | 1.332.0      |
| aws-eventstream           | 1.1.0        |
| aws-sigv4                 | 1.2.0        |
| aws-sdk-core              | 3.100.0      |
| aws-sdk-kms               | 1.34.1       |
| aws-sdk-s3                | 1.69.1       |
| nanaimo                   | 0.2.6        |
| colored2                  | 3.1.2        |
| claide                    | 1.0.3        |
| CFPropertyList            | 3.0.2        |
| atomos                    | 0.1.3        |
| xcodeproj                 | 1.16.0       |
| unicode-display_width     | 1.7.0        |
| public_suffix             | 4.0.5        |
| addressable               | 2.7.0        |
| tty-screen                | 0.8.0        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| babosa                    | 1.0.3        |
| excon                     | 0.75.0       |
| faraday                   | 1.0.1        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.1.7        |
| json                      | 2.3.0        |
| mini_magick               | 4.10.1       |
| dotenv                    | 2.7.5        |
| bundler                   | 2.1.4        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| multi_json                | 1.14.1       |
| signet                    | 0.14.0       |
| os                        | 1.1.0        |
| memoist                   | 0.16.2       |
| googleauth                | 0.13.0       |
| digest-crc                | 0.5.1        |
| google-cloud-errors       | 1.0.1        |
| google-cloud-env          | 1.3.2        |
| google-cloud-core         | 1.5.0        |
| google-cloud-storage      | 1.26.2       |
| openssl                   | 2.0.9        |
| io-console                | 0.4.6        |
| psych                     | 2.2.2        |
</details>

*generated on:* **2020-06-22**
 </pre>
</details>

**Comment by squallstar** [comment]
I am having the same issue using the latest on ` Spaceship::Tunes::Application.create`


```
app = Spaceship::Tunes::Application.create!(
  name: app_name,
  sku: bundle_id,
  bundle_id: bundle_id,
  company_name: "My company"
)
```

Error:

```
The request could not be completed because:
The Bundle ID you entered has already been used. The Bundle ID you entered has already been used.
```

**Comment by joshdholtz** [comment]
This is fixed in https://github.com/fastlane/fastlane/pull/16640 and is available for testing with `2.150.0.rc2` 😊

**Comment by gibertoni** [comment]
Thank you very much for the quick update

**Comment by joshdholtz** [comment]
Not a problem ❤️  Going to close this issue for now

**Comment by squallstar** [comment]
Thanks for the update! I'm going to give it a go now.

**Comment by squallstar** [comment]
@joshdholtz I've just tried and it still doesn't work.

```
Successfully installed fastlane-2.150.0.rc2
1 gem installed
root@ce643e521fa7:/usr/src/app# fastlane -v
fastlane installation at path:
/usr/local/bundle/gems/fastlane-2.150.0.rc2/bin/fastlane
-----------------------------
[✔] 🚀
fastlane 2.150.0.rc2
```

Running `Spaceship::Tunes::Application.create` with any bundle ID returns `The Bundle ID you entered has already been used.`

**Comment by joshdholtz** [comment]
@squallstar Hey! `Spaceship::Tunes` is what is broken. It has been removed completely in 2.150.0.rc2 and `Spaceship::ConnectAPI` is now replacing it. We had to switch to a whole new set of APIs and it wasn't possible to back support `Spaceship::Tunes` 😕

**Comment by squallstar** [comment]
@joshdholtz thanks, I'm going to try it out, hopefully it works. What about the following methods:

- Spaceship::Tunes.login(email, password)
- Spaceship::Tunes::Application.find(bundle_id)
- Spaceship::Tunes::Application.all.collect

Are they available in the new namespace and are there any docs I can follow to update them as required?

**Comment by joshdholtz** [comment]
@squallstar 

- `Spaceship::Tunes.login(email, password)` still exists (and will for a while) but will have an alias in `Spaceship::ConnectAPI` soon
- `Spaceship::ConnectAPI::App.find(bundle_id)`
  - Usage: https://github.com/fastlane/fastlane/blob/19e0a7b2279728086ade8f31a20f1bf02480736d/produce/lib/produce/itunes_connect.rb#L76
  - Implementation: https://github.com/fastlane/fastlane/blob/19e0a7b2279728086ade8f31a20f1bf02480736d/spaceship/lib/spaceship/connect_api/models/app.rb#L51-L55
- `Spaceship::ConnectAPI::App.all`
  - Implementation: https://github.com/fastlane/fastlane/blob/19e0a7b2279728086ade8f31a20f1bf02480736d/spaceship/lib/spaceship/connect_api/models/app.rb#L46-L49

**Comment by squallstar** [comment]
@joshdholtz I am getting the following error:

```
Traceback (most recent call last):
        1: from ./bin/fastlane/create-app.rb:35:in `<main>'
/usr/local/bundle/gems/fastlane-2.150.0.rc2/spaceship/lib/spaceship/connect_api/models/app.rb:65:in `create': undefined local variable or method `territory_ids' for Spaceship::ConnectAPI::App:Class (NameError)
        2: from ./bin/fastlane/create-app.rb:27:in `<main>'
        1: from ./bin/fastlane/create-app.rb:52:in `rescue in <main>'
./bin/fastlane/create-app.rb:52:in `*': no implicit conversion of String into Integer (TypeError)
```

This happens given the following source code:

```
app = Spaceship::ConnectAPI::App.create(
  name: app_name,
  sku: bundle_id,
  bundle_id: bundle_id
)
```

Is there any other new required parameter? Furthermore, the previous API required a `company_name` (also via `FASTLANE_ITC_TEAM_NAME`) which I believe was required when creating the very first app in the account. Is this not required anymore in the new API?

**Comment by joshdholtz** [comment]
@squallstar This is all fixed on `2.150.0.rc3` 😊 That `territory_ids` wasn’t supposed to be there.

**Comment by squallstar** [comment]
@joshdholtz I am still getting errors with the create method on the RC3:

```
Traceback (most recent call last):
        2: from ./bin/fastlane/create-app.rb:35:in `<main>'
        1: from /usr/local/bundle/gems/fastlane-2.150.0.rc3/spaceship/lib/spaceship/connect_api/models/app.rb:58:in `create'
/usr/local/bundle/gems/fastlane-2.150.0.rc3/spaceship/lib/spaceship/connect_api/tunes/tunes.rb:56:in `post_app': undefined method `each' for nil:NilClass (NoMethodError)
        2: from ./bin/fastlane/create-app.rb:27:in `<main>'
        1: from ./bin/fastlane/create-app.rb:52:in `rescue in <main>'
./bin/fastlane/create-app.rb:52:in `*': no implicit conversion of String into Integer (TypeError)
```

**Comment by squallstar** [comment]
@joshdholtz sorry to bother but is there any update on this? It is currently a blocker with no apparent workaround.

**Comment by joshdholtz** [comment]
@squallstar Hey 👋 Looks like you need to pass in a few more parameters

```rb
def post_app(name: nil, version_string: nil, sku: nil, primary_locale: nil, bundle_id: nil, platforms: nil)
```

```rb
app = Spaceship::ConnectAPI::App.create(
  name: app_name,
  version_string: “0.1.0”,
  sku: bundle_id,
  primary_locale: “en-US”,
  bundle_id: bundle_id,
  platforms: [
    Spaceship::ConnectAPI::Platform::IOS
  ]
)
```

I’m hoping to have some of the updated docs end of this week or next week with new `Spaceship::ConnectAPI`. Docs will be my focus once we get the official release out!

**Comment by squallstar** [comment]
@joshdholtz it's still not working, it returns a `An attribute value is invalid. - The bundleID is not available.` error given the input below:

```
app = Spaceship::ConnectAPI::App.create(
    name: 'Sample Nick',
    version_string: '1.0.0',
    sku: 'com.mysample.foobar123123abc',
    primary_locale: 'en-US',
    bundle_id: 'com.mysample.foobar123123abc',
    platforms: [
      Spaceship::ConnectAPI::Platform::IOS
    ]
  )
```

**Comment by joshdholtz** [comment]
@squallstar That is an error that is received from App Store Connect. It looks like the bundle identifier you are trying to use is already taken by somebody else. If you try to create a new app with that exact same bundle identifier on the App Store Connect website, you would get that same error.

**Comment by squallstar** [comment]
@joshdholtz I tried literally a dozen of random bundles and I am receiving the same error so it can't be an issue related to bundles being in use.

**Comment by joshdholtz** [comment]
@squallstar Did you create the app on the Developer Center/Portal? `Spaceship::ConnectAPI::App.create` only creates it in AppStoreConnect.

The Developer Center/Portal is a different resource. I think that code still needs to be added into the `Spaceship::ConnectAPI` module.

**Comment by squallstar** [comment]
@joshdholtz I previously never had to manually create the app on the dev portal. Is this now required?

To confirm, I have used this in the past but only to produce enterprise/in-house builds, not App Store ones:

```
 app = Spaceship::Portal.app.create!(
    name: app_name,
    bundle_id: bundle_id
  )
```

Can you confirm
1. If the app should now first be created on the developer centre via the above code
2. What changes should be made to `Spaceship::Portal.app.create` if required

**Comment by joshdholtz** [comment]
@squallstar `Spaceship::Portal` is Developer Center portal. This would not have created the app on App Store Connect.

`Spaceship::ConnectAPI::App` is for App Store Connect but requires the bundle identifier to be be created first. I don’t have this model yet but I will add `Spaceship::ConnectAPI::BundleID`

In the mean time...


https://github.com/fastlane/fastlane/blob/master/produce/lib/produce/developer_center.rb#L59-L62

This should still work for creating the app on the Developer Center before you create it on App Store Connect.

**Comment by squallstar** [comment]
@joshdholtz I'm still confused at whether the change to be made is to create the app on the developer centre first, because this isn't the case for the current (not working anymore) version of Fastlane. I have created hundreds of apps over the years on the App Store Connect without explicitly having to create them in the Dev Portal. Has this changed now?

**Comment by joshdholtz** [comment]
@squallstar I think I’m confused on why you are confused 😊 So lets start over!

So in the past, what code have you been using to create your apps? Were you only using this? 👇

```rb
 app = Spaceship::Portal.app.create!(
    name: app_name,
    bundle_id: bundle_id
  )
```

**Comment by squallstar** [comment]
@joshdholtz I used the `Spaceship::Tunes::Application.create` method only for App Store apps.

On the other hand, for unsigned or enterprise (in house) builds I used `Spaceship::Portal.app.create`.

I never had to use the Portal APIs for App Store builds.

**Comment by joshdholtz** [comment]
@squallstar That _may_ have been a feature that was in the legacy API (which is no longer available for us to you)... where it created it on both the portal and app store for you.

Now you will need to first create it on the developer portal and then create it on app store after that (if not enterprise).

I will try to get those APIs added for you today with an example and then mention you in the PR.

**Comment by squallstar** [comment]
@joshdholtz I see, thanks I appreciate the help so far. I'll be updating my code as suggested and give it a go. Keep me posted on the required APIs, thanks again.

**Comment by squallstar** [comment]
@joshdholtz sorry to bother but I am getting another error, this time is the `app.details` method that is undefined:

```
app = Spaceship::ConnectAPI::App.find(bundle_id)

app.details..primary_category
```

Error: `undefined method `details' for #<Spaceship::ConnectAPI::App:0x000055b9e4258138>`

**Comment by joshdholtz** [comment]
@squallstar Im just waking up for the day but I'll take a look in about 45 minutes!

**Comment by squallstar** [comment]
For clarity, this is the list of attributes I was using with the previous version:

```
details = app.details

result = {
    :primary_category => details.primary_category,
    :primary_first_sub_category => details.primary_first_sub_category,
    :primary_second_sub_category => details.primary_second_sub_category,
    :secondary_category => details.secondary_category,
    :secondary_first_sub_category => details.secondary_first_sub_category,
    :secondary_second_sub_category => details.secondary_second_sub_category,
    :primary_locale_code => details.primary_locale_code,
    :available_primary_locale_codes => details.available_primary_locale_codes,
    :languages => details.languages
}
```

I'm looking at https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/connect_api/models/app.rb but can't manage to find those attributes.

**Comment by joshdholtz** [comment]
@squallstar Still on my phone but things changed a lot. I'll be getting documentation together next week of the mew mode/resource relationships. 

I'll link you to example or paste some code for you in about 30 minutes.

**Comment by squallstar** [comment]
Thanks, I'm keeping an eye on this thread. These are also other attributes I've been using:

```
live_version = app.live_version
submission = app.edit_version

result = {
  :issues => {
    :count => app.issues_count,
    :error => app.resolution_center["value"],
    :notes => app.resolution_center["appNotes"]
  },
  :live => {
    :status => live_version ? live_version.app_status : nil,
    :version => live_version ? live_version.version : nil
  },
  :submission => submission ? {
    :status => submission ? submission.app_status : nil,
    :version => submission ? submission.version : nil
  } : nil
}
```

```
apps = Spaceship::Portal.app.all.collect

apps = apps.map {|a| {
  :id => a.app_id,
  :name => a.name,
  :bundle => a.bundle_id
} }
```

```
apps = Spaceship::ConnectAPI::App.all.collect

apps = apps.map {|a| {
  :id => a.apple_id,
  :name => a.name,
  :bundle => a.bundle_id
} }
```

**Comment by joshdholtz** [comment]
Update: I'm waiting on a forever macOS update so still not on my computer 😛

**Comment by squallstar** [comment]
@joshdholtz I feel the pain, my Mac always tries to update in the worst possible situations like when I have a meeting in 5 minutes 😂

**Comment by joshdholtz** [comment]
@squallstar Here you go 👉 https://github.com/fastlane/fastlane/pull/16741

I'll push a new release out later today with this

**Comment by squallstar** [comment]
Thanks! I'll have a go on Monday morning.
